src/model.rs

```diff
@@ -46,6 +46,32 @@ pub(crate) struct SavedState {
     pub(crate) pending: Option<BTreeMap<String, ContentMetadata>>,
 }
 
+/// Representation of bootupd's worldview at a point in time.  This is intended
```

Comment:
Nit: it is customary to put a newline between the starting line and the rest of the text. That way `rustdocs` knows how to split the short summary and the longer-form description.